### DISCUSSION TITLE
Add ability for scenarios to allow more months (or remove months) in a year

### DISCUSSION THREAD

**Discussion by Generalcamo** [OpenRCT2/OpenRCT2#18281] [Feature requests]
Roller Coaster Tycoon traditionally sets its year from March 1st to October 31st. This is a general "Amusement Park Season" for the northern hemisphere in the northern latitudes, which is where the original RCT is mostly set. This assumption breaks down once you leave these latitudes however:

- Near the Equator, the amusement park season is year-round as there isn't a harsh winter preventing Roller Coasters from operating safely
- In the Southern Hemisphere, this season is reversed (skipping May, June, July, and August) as Winter is inverted from the Northern Hemisphere

To allow for more diverse locations, the ability to set which months the park is open should be allowed for custom scenarios. To avoid breaking compatibility, the default should remain what the original RCT had (March to October).

A complication is how to handle February. Under the Gregorian calendar, February has 28 days except for leap years, where it has 29 days. A leap year is every 4 years, skipping every 100 years except if that year is divisible by 400 (So 1900 was not a leap year, 2000 was). As Roller Coaster Tycoon does not use a standard calendar year, this poses an issue for how to handle this.

**Comment by ocalhoun6** [comment]
A good point. Even in North America, parks in the south tend to stay open all year long.

-------

>A complication is how to handle February. Under the Gregorian calendar, February has 28 days except for leap years, where it has 29 days. A leap year is every 4 years

Since RCT years don't have to match up with actual astronomical cycles, it would probably be best to just _always_ let February have 28 days, no matter what.

(Anyway, if I remember right, all the months in RCT2 take exactly the same amount of real-world time, regardless of whether they're 30 or 31 days. In 31-day months, the days are slightly shorter in order to squeeze them all in. So it doesn't really matter whether February has 28 or 29 days -- it's the same amount of time regardless.)

---------

Also would make for a fun scenario restriction if you could make the opening season even shorter, like for a park at a high latitude, where it might not be able to stay open for as much of the year as a more temperate park.

That effectively makes the scenario deadline shorter.

**Comment by TheClaudeQC** [comment]
I really like that Idea, So I could see some parks, probably like magic mountain or texas, having it. but texas could be shorten by 1 year to have a similar simulation, but instread of october, it could be 31st of July

**Comment by ZeeMaji** [comment]
With the recent introduction of climate objects, i think this could be integrated into them as a new option. I think this would make more sense than a separate park option as the climate is a big part of the operating seasons of real world parks and it would avoid some weird issues that could pop up if year duration was separate as you could enable monthes that the current climate object doesn't have climate data for.
